packages/mui-material/src/styles/ThemeProviderWithVars.tsx

```diff
@@ -2,20 +2,18 @@
 import * as React from 'react';
 import { unstable_createCssVarsProvider as createCssVarsProvider, SxProps } from '@mui/system';
 import styleFunctionSx from '@mui/system/styleFunctionSx';
-import extendTheme, { SupportedColorScheme, CssVarsTheme } from './extendTheme';
+import createThemeWithVars, { SupportedColorScheme, CssVarsTheme } from './createThemeWithVars';
 import createTypography from './createTypography';
 import THEME_ID from './identifier';
 import { defaultConfig } from '../InitColorSchemeScript/InitColorSchemeScript';
 
-const defaultTheme = extendTheme();
-
 const {
   CssVarsProvider,
   useColorScheme,
   getInitColorSchemeScript: deprecatedGetInitColorSchemeScript,
 } = createCssVarsProvider<SupportedColorScheme, typeof THEME_ID>({
   themeId: THEME_ID,
-  theme: defaultTheme,
+  theme: createThemeWithVars,
```

Comment:
What's the motivation from changing from defaultTheme to a create theme function usage? Do we plan to have two default themes, one with vars and one without?